File: csharp/resistor-color-trio/ResistorColorTrio.cs
public static class ResistorColorTrio
{
    public enum ResistorColor
    {
        Black = 0,
        Brown = 1,
        Red = 2,
        Orange = 3,
        Yellow = 4,
        Green = 5,
        Blue = 6,
        Violet = 7,
        Grey = 8,
        White = 9,
    }
    private static ulong GetResistorColorValue(string color)
    => (ulong)Enum.Parse<ResistorColor>(color, true);
    public static string Label(string[] colors)
    {
        if (colors.Length < 3) throw new ArgumentException("Not Trio Resistor");
        ulong value = GetResistorColorValue(colors[0]) * 10 + GetResistorColorValue(colors[1]);
        value *= (uint) Math.Pow(10, GetResistorColorValue(colors[2]));

        return value switch
        {
            > (ulong)1e9 => $"{value / (int)1e9} gigaohms",
            > (ulong)1e6 => $"{value / (int)1e6} megaohms",
            > (ulong)1e3 => $"{value / (int)1e3} kiloohms",
            _ => $"{value} ohms"
        };
    }
}

File: javascript/resistor-color-trio/resistor-color-trio.js
//
// This is only a SKELETON file for the 'Resistor Color Trio' exercise. It's been provided as a
// convenience to get you started writing code faster.
//
const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white'
];

export class ResistorColorTrio {
  constructor(colors) {
    this.colors = colors;
  }

  get label() {
    if (this.colors.some(color => COLORS.indexOf(color) === -1)){
      throw new Error(`invalid color`);
    }
    const [a, b, c] = this.colors.map(color => COLORS.indexOf(color));
    let value = (a * 10 + b) * (10 ** c); //(firstDigit * 10 + secondDigit) * 10^multiplier

    let unit = `ohms`;

    if (value >= 1_000_000_000){
      value /= 1_000_000_000;
      unit = `gigaohms`;
    } else if (value >= 1_000_000){
      value /= 1_000_000;
      unit = `megaohms`;
    } else if (value >= 1_000){
      value /= 1_000;
      unit = `kiloohms`;
    }

    return `Resistor value: ${value} ${unit}`;
  }
}
